File: Exercises/Ex1/part4.go
package main
	
import (
	"fmt"
	"time"
	"runtime"
)

var i = 0

func thread1(){
	for j:= 0; j <10; j++{
		i+=1
	}
}

func thread2(){
	for j:= 0; j <10; j++{
		i-=1
	}
}

func main(){
	runtime.GOMAXPROCS(runtime.NumCPU(runtime.NumCPU())


	go thread1()
	go thread2()
	time.Sleep(1000*time.Millisecond)
	fmt.Println(i)
}

File: Exercises/Ex1/part4.py
import thread,time


i = 0

def main():

	
	global i
	thread.start_new_thread(thread1,())
	thread.start_new_thread(thread2,())
	time.sleep(10)
	print "nubmer", i


def thread1():
	global i
	for j in range(1,1000000):
		i+=1

def thread2():
	global i
	for j in range (1,1000000):
		i-=1

main()